src/schema-functions.ts

```diff
@@ -571,6 +669,47 @@ namespace validation {
       .filter(notEmpty);
   };
 
+  export const validateForeignKey: CrossSchemaValidationFunction = (
+    schemaDef: SchemaDefinition,
+    schemasData: Record<string, SchemaData>
+  ) => {
+    const errors: Array<SchemaValidationError> = [];
+    const foreignKeyDefinitions = schemaDef?.restrictions?.foreignKey;
+    if (foreignKeyDefinitions) {
+      foreignKeyDefinitions.forEach(foreignKeyDefinition => {
+        const localSchemaData = schemasData[schemaDef.name];
+        const foreignSchemaData = schemasData[foreignKeyDefinition.schema];
```

Comment:
`localSchemaData` and `foreignSchemaData` could be `undefined` by accessing their values using brackets even though TS does not indicate this.  We should make sure that each of these constants is at least set to be an empty array if no data records are provided for this schema:
```suggestion
        const localSchemaData = schemasData[schemaDef.name] || [];
        const foreignSchemaData = schemasData[foreignKeyDefinition.schema] || [];
```